service/gradle/dependencies.gradle

```diff
@@ -42,24 +42,24 @@ dependencies {
   implementation group: "org.glassfish.jersey.inject", name: "jersey-hk2"
 
   // OpenTelemetry utilities for various HTTP clients
-  var openTelemetryVersion = '1.32.0'
+  var openTelemetryVersion = '1.34.1'
+  var openTelemetryInstrumentationVersion = '2.0.0'
   implementation "io.opentelemetry:opentelemetry-api:${openTelemetryVersion}"
   implementation "io.opentelemetry:opentelemetry-sdk:${openTelemetryVersion}"
   implementation "io.opentelemetry:opentelemetry-sdk-common:${openTelemetryVersion}"
   implementation "io.opentelemetry:opentelemetry-sdk-metrics:${openTelemetryVersion}"
   implementation "io.opentelemetry:opentelemetry-exporter-logging:${openTelemetryVersion}"
-  implementation "io.opentelemetry.semconv:opentelemetry-semconv:1.21.0-alpha"
-  implementation "io.opentelemetry.instrumentation:opentelemetry-spring-webmvc-6.0:${openTelemetryVersion}-alpha"
-  implementation "io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:${openTelemetryVersion}"
-  implementation "io.opentelemetry.instrumentation:opentelemetry-spring-boot:${openTelemetryVersion}-alpha"
+  implementation "io.opentelemetry.instrumentation:opentelemetry-spring-webmvc-6.0:${openTelemetryInstrumentationVersion}-alpha"
+  implementation "io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:${openTelemetryInstrumentationVersion}"
+  implementation "io.opentelemetry.instrumentation:opentelemetry-spring-boot:${openTelemetryInstrumentationVersion}-alpha"
 
-  var gcpOpenTelemetryExporterVersion = '0.25.2'
+  // Google cloud open telemetry exporters
+  var gcpOpenTelemetryExporterVersion = '0.27.0'
   implementation "com.google.cloud.opentelemetry:exporter-trace:${gcpOpenTelemetryExporterVersion}"
   implementation "com.google.cloud.opentelemetry:exporter-metrics:${gcpOpenTelemetryExporterVersion}"
```

Comment:
Naively mimicking what I saw in https://github.com/DataBiosphere/terra-common-lib/pull/131/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7 -- but I want to understand whether there's a better way deal with these transitive deps.

Reply:
@blakery this was a funny one.  There is an opentelemetry-bom (bill of materials) that ideally we'd use to version OTEL dependencies rather than versioning them directly, so that they remain in compatibility with one another.  I was surprised to find that this is a dependency that Spring Boot Dependency Manager brings in, and that plug-in clobbers the versions we see from other sources.

Once I instructed the Spring Boot dependency manager to use an OTEL BOM version compatible with latest TCL, I was able to dramatically simplify our OTEL direct dependencies:

https://github.com/DataBiosphere/terra-workspace-manager/pull/1666/commits/8aef8d631b7f853a0f6d14e7120b068031e03a70